catalog/ftp/src/main/java/ddf/catalog/ftp/ftplets/FtpRequestHandler.java

```diff
@@ -109,6 +108,7 @@ public FtpletResult onUploadStart(FtpSession session, FtpRequest request)
 
         Subject shiroSubject = (Subject) session.getAttribute(SUBJECT);
         if (shiroSubject != null) {
```

Comment:
you could reduce complexity by checking if(shiroSubject==null) instead. Then you wouldn't have to wrap this code block in an if statement